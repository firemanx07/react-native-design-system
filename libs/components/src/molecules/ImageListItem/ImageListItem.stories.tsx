import { ComponentStory, ComponentMeta } from '@storybook/react';
import React from 'react';
import { Image } from 'react-native';

import { Illustration, IllustrationName, IllustrationSize } from '../../atoms';
import { StorybookScreen, StorybookSection } from '../../storybook';
import {
  ImageListItem as ImageListItemBase,
  PropsType as ImageListItemProps,
} from './ImageListItem';

const Story: ComponentMeta<typeof ImageListItemBase> = {
  component: ImageListItemBase,
  title: 'Molecules/Image List Item',
};
export default Story;

const defaultArgs = {
  title: 'Recently added',
  subtitle: '300 files',
  hasArrow: true,
  renderImage: (size: number) => (
    <Image
      style={{ width: size, height: size }}
      source={{ uri: 'https://picsum.photos/64/64' }}
    />
  ),
};

const Template: ComponentStory<typeof ImageListItemBase> = (
  args: ImageListItemProps,
) => (
  <StorybookScreen title="ImageListItem:">
    <StorybookSection>
      <ImageListItemBase {...args} />
    </StorybookSection>
    <StorybookSection title="with Illustration">
      <ImageListItemBase
        {...defaultArgs}
        hasArrow={false}
        renderImage={() => (
          <Illustration
            name={IllustrationName.Cards}
            size={IllustrationSize.S}
          />
        )}
      />
    </StorybookSection>
    <StorybookSection title="with image">
      <ImageListItemBase {...defaultArgs} hasArrow={false} />
    </StorybookSection>
  </StorybookScreen>
);

const parameters = {
  controls: {
    include: ['title', 'subtitle', 'hasArrow'],
  },
};

export const ImageListItem = Template.bind({});
ImageListItem.parameters = parameters;
ImageListItem.args = {
  ...defaultArgs,
};
