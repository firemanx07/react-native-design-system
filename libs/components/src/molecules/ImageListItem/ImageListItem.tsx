import { styled, useNamespacedTheme } from '@proxym/themes';
import React, { memo, ReactNode } from 'react';
import { ViewStyle, StyleProp } from 'react-native';

import { BaseText, ForwardIcon, TextVariant, TextWeight } from '../../atoms';
import { TestIDType } from '../../types';

export type PropsType = {
  title: string;
  renderImage: (size: number) => ReactNode;
  hasArrow?: boolean;
  subtitle?: string;
  style?: StyleProp<ViewStyle>;
} & TestIDType;

const IMAGE_SIZE = 64;

export const ImageListItem = ({
  title,
  subtitle,
  hasArrow,
  renderImage,
  style,
  testID,
}: PropsType) => {
  const { iconSize, colors } = useNamespacedTheme();
  return (
    <Container style={style} testID={testID}>
      <ImageWrapper>{renderImage(IMAGE_SIZE)}</ImageWrapper>
      <TextWrapper>
        <BaseText
          variant={TextVariant.L}
          weight={TextWeight.Bold}
          numberOfLines={2}
        >
          {title}
        </BaseText>
        {!!subtitle && (
          <Subtitle
            variant={TextVariant.M}
            weight={TextWeight.SemiBold}
            numberOfLines={2}
          >
            {subtitle}
          </Subtitle>
        )}
      </TextWrapper>
      {hasArrow && (
        <ForwardIcon
          width={iconSize.medium}
          height={iconSize.medium}
          fill={colors.dark}
        />
      )}
    </Container>
  );
};

const Container = styled.View`
  flex-direction: row;
  align-items: center;
`;

const ImageWrapper = styled.View`
  background-color: ${({ theme }) => theme.ds.colors.secondary};
  border-radius: 4px;
  margin-right: ${({ theme }) => theme.ds.spacing.primary}px;
  width: ${IMAGE_SIZE}px;
  height: ${IMAGE_SIZE}px;
  overflow: hidden;
  justify-content: center;
  align-items: center;
`;

const TextWrapper = styled.View`
  flex: 1;
  padding-right: ${({ theme }) => theme.ds.spacing.small}px;
`;

const Subtitle = styled(BaseText)`
  color: ${({ theme }) => theme.ds.colors.grayDark};
`;

export default memo(ImageListItem);
