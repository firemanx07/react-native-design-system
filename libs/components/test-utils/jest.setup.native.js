import { Animated } from 'react-native';

require('react-native-reanimated/lib/reanimated2/jestUtils').setUpTests();
require('react-native-gesture-handler/jestSetup.js');

console.warn = () => {
  //this is mock
};

Animated['timing'] = (value, config) => ({
  start: callback => {
    value.setValue(config.toValue);
    callback && callback({ finished: true });
  },
  stop: jest.fn(),
});
