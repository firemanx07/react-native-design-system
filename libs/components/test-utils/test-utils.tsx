import { theme, ThemeProvider } from '@celebrate-app/themes';
import { Story } from '@storybook/react';
import {
  cleanup,
  render,
  RenderOptions,
  waitFor,
} from '@testing-library/react-native';
import React, { ReactNode } from 'react';

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions
) => {
  return render(ui, { wrapper: Providers, ...options });
};

beforeEach(cleanup);
afterEach(cleanup);

export function testStory(displayName: string, TestComponent: Story) {
  test(`${displayName} snapshot test`, async () => {
    const comp = customRender(<TestComponent {...TestComponent.args} />);
    jest.runOnlyPendingTimers();

    await waitFor(() => {
      expect(comp).toBeTruthy();
      expect(comp.toJSON()).toMatchSnapshot();
    });
  });
}

export function testComponentSnapshot(
  TestComponent: React.ReactElement<
    any,
    string | React.JSXElementConstructor<any>
  >
) {
  test('Snapshot test', () => {
    const comp = customRender(TestComponent);
    expect(comp).toBeTruthy();
    expect(comp.toJSON()).toMatchSnapshot();
  });
}

export function testAllStories(stories: any) {
  Object.keys(stories).forEach((story) => {
    if (story !== 'default') {
      testStory(story, stories[story]);
    }
  });
}
