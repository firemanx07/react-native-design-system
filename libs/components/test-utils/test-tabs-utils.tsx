import { waitFor } from '@testing-library/react-native';

import { customRender } from './test-utils';

type renderTabsProps = {
  elem: JSX.Element;
  testIDs: string[];
};
export const renderTabs = ({ elem, testIDs }: renderTabsProps) => {
  const comp = customRender(elem);
  const tabs: ReturnType<typeof comp.getByTestId>[] = [];
  testIDs.forEach(testID => tabs.push(comp.getByTestId(testID)));
  return { comp, tabs };
};
export const testTabs = async (args: ReturnType<typeof renderTabs>) => {
  await waitFor(() => {
    expect(args.comp).toBeTruthy();
    args.tabs.forEach(tab => expect(tab).toBeDefined());
  });
};
