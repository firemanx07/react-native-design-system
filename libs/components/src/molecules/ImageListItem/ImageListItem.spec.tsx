import { testAllStories } from '../../../test-utils';
import * as stories from './ImageListItem.stories';

describe('ImageListItem tests', () => {
  testAllStories(stories);
});
