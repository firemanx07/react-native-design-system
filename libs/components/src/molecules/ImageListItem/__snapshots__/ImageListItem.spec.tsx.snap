// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageListItem tests ImageListItem snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <Text
    fontFamily="Rubik-Bold"
    fontSize={20}
    lineHeight={26}
    style={
      [
        {
          "color": "#2C2C2C",
          "fontFamily": "Rubik-Bold",
          "fontSize": 20,
          "lineHeight": 26,
        },
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    ImageListItem:
  </Text>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#49C4E9",
              "borderRadius": 4,
              "height": 64,
              "justifyContent": "center",
              "marginRight": 16,
              "overflow": "hidden",
              "width": 64,
            },
          ]
        }
      >
        <Image
          source={
            {
              "uri": "https://picsum.photos/64/64",
            }
          }
          style={
            {
              "height": 64,
              "width": 64,
            }
          }
        />
      </View>
      <View
        style={
          [
            {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "paddingRight": 8,
            },
          ]
        }
      >
        <Text
          fontFamily="Rubik-Bold"
          fontSize={16}
          lineHeight={20}
          numberOfLines={2}
          style={
            [
              {
                "color": "#2C2C2C",
                "fontFamily": "Rubik-Bold",
                "fontSize": 16,
                "lineHeight": 20,
              },
            ]
          }
        >
          Recently added
        </Text>
        <Text
          fontFamily="Rubik-SemiBold"
          fontSize={14}
          lineHeight={20}
          numberOfLines={2}
          style={
            [
              {
                "color": "#2C2C2C",
                "fontFamily": "Rubik-SemiBold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#C7C7D2",
              },
            ]
          }
        >
          300 files
        </Text>
      </View>
      <test-file-stub
        fill="#2C2C2C"
        height={20}
        width={20}
      />
    </View>
  </View>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <Text
      fontFamily="Rubik-Bold"
      fontSize={16}
      lineHeight={20}
      style={
        [
          {
            "color": "#2C2C2C",
            "fontFamily": "Rubik-Bold",
            "fontSize": 16,
            "lineHeight": 20,
          },
          {
            "backgroundColor": "#C7C7D2",
            "left": 16,
            "paddingHorizontal": 8,
            "position": "absolute",
            "top": -12,
          },
        ]
      }
    >
      with Illustration
    </Text>
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#49C4E9",
              "borderRadius": 4,
              "height": 64,
              "justifyContent": "center",
              "marginRight": 16,
              "overflow": "hidden",
              "width": 64,
            },
          ]
        }
      >
        <Image
          height={48}
          resizeMode="contain"
          source={
            {
              "uri": "",
            }
          }
          style={
            [
              {
                "height": 48,
                "width": 60,
              },
            ]
          }
          width={60}
        />
      </View>
      <View
        style={
          [
            {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "paddingRight": 8,
            },
          ]
        }
      >
        <Text
          fontFamily="Rubik-Bold"
          fontSize={16}
          lineHeight={20}
          numberOfLines={2}
          style={
            [
              {
                "color": "#2C2C2C",
                "fontFamily": "Rubik-Bold",
                "fontSize": 16,
                "lineHeight": 20,
              },
            ]
          }
        >
          Recently added
        </Text>
        <Text
          fontFamily="Rubik-SemiBold"
          fontSize={14}
          lineHeight={20}
          numberOfLines={2}
          style={
            [
              {
                "color": "#2C2C2C",
                "fontFamily": "Rubik-SemiBold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#C7C7D2",
              },
            ]
          }
        >
          300 files
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <Text
      fontFamily="Rubik-Bold"
      fontSize={16}
      lineHeight={20}
      style={
        [
          {
            "color": "#2C2C2C",
            "fontFamily": "Rubik-Bold",
            "fontSize": 16,
            "lineHeight": 20,
          },
          {
            "backgroundColor": "#C7C7D2",
            "left": 16,
            "paddingHorizontal": 8,
            "position": "absolute",
            "top": -12,
          },
        ]
      }
    >
      with image
    </Text>
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#49C4E9",
              "borderRadius": 4,
              "height": 64,
              "justifyContent": "center",
              "marginRight": 16,
              "overflow": "hidden",
              "width": 64,
            },
          ]
        }
      >
        <Image
          source={
            {
              "uri": "https://picsum.photos/64/64",
            }
          }
          style={
            {
              "height": 64,
              "width": 64,
            }
          }
        />
      </View>
      <View
        style={
          [
            {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "paddingRight": 8,
            },
          ]
        }
      >
        <Text
          fontFamily="Rubik-Bold"
          fontSize={16}
          lineHeight={20}
          numberOfLines={2}
          style={
            [
              {
                "color": "#2C2C2C",
                "fontFamily": "Rubik-Bold",
                "fontSize": 16,
                "lineHeight": 20,
              },
            ]
          }
        >
          Recently added
        </Text>
        <Text
          fontFamily="Rubik-SemiBold"
          fontSize={14}
          lineHeight={20}
          numberOfLines={2}
          style={
            [
              {
                "color": "#2C2C2C",
                "fontFamily": "Rubik-SemiBold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#C7C7D2",
              },
            ]
          }
        >
          300 files
        </Text>
      </View>
    </View>
  </View>
</View>
`;
